### ISSUE TITLE
Pods Admin/- Template edit screen does not display all fields the Songs pod

### ISSUE THREAD

**Issue by dlincoln** [pods-framework/pods#3074]
When I create a new template and set Pod Reference to my "Songs" pod in the edit window, mostly native WP fields show up, alongside external related fields, with only some of the added custom fields in the Pod.

If I set the Pod Reference to Songs first, before selecting a smaller pod, it displays a truncated list of unrelated generic fields in it's pick-list. Selecting another Pod in this field makes things go back to normal.

I thought this might be related to other templates somehow, so I killed all of my older ones to try again. No love.

**Comment by dlincoln** [comment]
![song pod fields as seen by template](https://cloud.githubusercontent.com/assets/13316833/8863098/a0b9dc98-3163-11e5-9dfa-ead7eb67c6c4.png)

**Comment by dlincoln** [comment]
https://gist.github.com/dlincoln/81c92e96625bcd4a0cf0#file-more-20templates-json

**Comment by dlincoln** [comment]
Overnight I slept my computer, woke it up with myself and returned to the page above. This morning the same pod displays the proper fields! No other changes on my end (site untouched):
![song pod fields seen on friday](https://cloud.githubusercontent.com/assets/13316833/8875905/39ebf212-31ea-11e5-812a-8a946adb99a2.png)

**Comment by dlincoln** [comment]
On a new, clean local installation (based on the attached gist), minus many plugs and themes, I went to Pods Admin/Templates, and add-new. I named the template, and tried to set a reference. It displayed the truncated reference list (as in the first image above) for the Song field. I checked another field, same display (basic WP elements, no custom Pods stuff showing).

I hit "save draft", and WP churned away, throwing the blank, no-content screen originally reported. New symptom: It threw the following WP debug error as well on this installation:

Fatal error: Allowed memory size of 268435456 bytes exhausted (tried to allocate 72 bytes) in /Users/dlincoln/Documents/Websites/musicli-dev.dev/wp-content/plugins/pods/components/Templates/includes/functions-pod_reference.php on line 89
Call Stack
#
Time
Memory
Function
Location
1
0.0012
373976
{main}( )
../post.php:0
2
0.5086
49144184
include( '/Users/dlincoln/Documents/Websites/musicli-dev.dev/wp-admin/edit-form-advanced.php' )
../post.php:212
3
0.7268
50645384
do_meta_boxes( )
../edit-form-advanced.php:607
4
0.7435
50650760
call_user_func:{/Users/dlincoln/Documents/Websites/musicli-dev.dev/wp-admin/includes/template.php:1074} ( )
../template.php:1074
5
0.7435
50651088
Pods_Templates_Frontier->render_metaboxes_custom( )
../template.php:1074
6
0.7442
50661440
include( '/Users/dlincoln/Documents/Websites/musicli-dev.dev/wp-content/plugins/pods/components/Templates/includes/element-pod_reference.php' )
../class-pods_templates.php:241
7
0.7731
52916216
pq_loadpod( )
../element-pod_reference.php:22
8
0.7734
52932016
pq_tunnel_pod_field( )
../functions-pod_reference.php:30
9
0.7736
52949472
pq_tunnel_pod_field( )
../functions-pod_reference.php:89
10
4.6490
240398856
array_merge ( )
../functions-pod_reference.php:89

**Comment by dlincoln** [comment]
ISSUE IDENTIFIED: I stripped down a local installation to nothing but Pods and Twenty Fifteen. Same errors. I killed a few big Pods which threw even more errors... a clue!

I killed EVERY Pod CPT but one, leaving my copious taxonomies in place. I then removed all relationships from the remaining pod, turning them into text. When I returned to Add Template, it worked as advertised.

So the bug is connected to having too many relationships, creating memory problems.

Workaround: Use all flat tables, or edit PHP as external templates. Ick.

**Comment by quasel** [comment]
well my debug bar says 24 MB must be something diffrent 

WARNING: wp-content/plugins/pods/components/Templates/includes/element-pod_reference.php:14 - Illegal string offset 'pod'
include('wp-admin/edit-form-advanced.php'), do_meta_boxes, call_user_func, Pods_Templates_Frontier->render_metaboxes_custom, include('/plugins/pods/components/Templates/includes/element-pod_reference.php')

**Comment by dlincoln** [comment]
Yes, I think they may be related though: ajax is mangling things (or exhausting itself) on the way out.

**Comment by jamesgol** [comment]
I think this was fixed by #4000
